### ISSUE TITLE
cpp_classification compile error

### ISSUE THREAD

**Issue by ProgmasterX** [BVLC/caffe#4282]
Line 162:  
net_->Forward();

error: no matching function for call to ‘caffe::Net<float>::Forward(caffe::Blob<float>*&)’
   net_->Forward(input_layer);
                            ^
main.cpp:166:28: note: candidates are:
In file included from /usr/include/caffe/caffe.hpp:12:0,
                 from main.cpp:10:
/usr/include/caffe/net.hpp:53:31: note: const std::vector<caffe::Blob<Dtype>*>& caffe::Net<Dtype>::Forward(const std::vector<caffe::Blob<Dtype>*>&, Dtype*) [with Dtype = float]
   const vector<Blob<Dtype>*>& Forward(const vector<Blob<Dtype>* > & bottom,
                               ^
/usr/include/caffe/net.hpp:53:31: note:   no known conversion for argument 1 from ‘caffe::Blob<float>*’ to ‘const std::vector<caffe::Blob<float>*, std::allocator<caffe::Blob<float>*> >&’
/usr/include/caffe/net.hpp:59:10: note: std::string caffe::Net<Dtype>::Forward(const string&, Dtype*) [with Dtype = float; std::string = std::basic_string<char>]
   string Forward(const string& input_blob_protos, Dtype* loss = NULL);
          ^
/usr/include/caffe/net.hpp:59:10: note:   no known conversion for argument 1 from ‘caffe::Blob<float>*’ to ‘const string& {aka const std::basic_string<char>&}’
make: *** [all] Error 1

**Comment by flx42** [comment]
Looks like an installation problem, or you modified the `cpp_classification` example. Next time, ask your question on the `caffe-users` list.
From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:

>     Please do not post usage, installation, or modeling questions, or other requests for help to Issues.
>     Use the caffe-users list instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.

**Comment by seanbell** [comment]
Please ask usage questions on the mailing list.

From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:
> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
